### PR TITLE
Migrate to new `Plugin` API

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR migrates Validation plugins to new `Plugin` API recently introduced in ProtoData.

Convenience extension functions `asA()` and `asB()` were also adopted.

### CODE REVIEW

File: model/src/main/java/io/spine/validation/ValidationPolicy.java [hunk 1]
```diff
@@ -48,25 +48,24 @@ public abstract class ValidationPolicy<E extends EventMessage>
...
+     *               return ignoring();
      *           }
      *           return myCustomRule(event);
      *       }
      * </pre>
      */
-    protected final EitherOf2<RuleAdded, Nothing> noReaction() {
-        //TODO:2024-08-11:alexander.yevsyukov: Use EventProducer.noReaction() extension from `core-java`.
-        return EitherOf2.withB(nothing());
+    protected final EitherOf2<RuleAdded, NoReaction> ignoring() {
```
Comment (armiol) [inline model/src/main/java/io/spine/validation/ValidationPolicy.java:68]: See my comments above. And let's discuss.

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov please see my comments, and let's discuss.

**Review by yevhenii-nadtochii — APPROVED** [review]
LGTM from my side.

I would also go with `noReaction()` because it matches the method signature of the reactor.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM.

@yevhenii-nadtochii FYI. We have settled on `ignore()`, as `noReaction()` is impossible to have to return `Either...` instance — in the same class hierarchy it is already busy returning plain `NoReaction`. So there's that.

**Pull request merged**